src/main/java/com/jfrog/ide/common/yarn/YarnTreeBuilder.java

```diff
@@ -93,6 +90,120 @@ private void addDepTreeNodes(Map<String, DepTreeNode> nodes, JsonNode jsonDep, D
         dependenciesList.elements().forEachRemaining(subDep -> addDepTreeNodes(nodes, subDep, depNode));
     }
 
+    private boolean isDirectDependency(String rawDependency) {
+        rawDependency = StringUtils.lowerCase(rawDependency); // The word specified can be in upper or lower case
+        // If rawDependency contains "specified in" or "workspace-aggregator-" it is a direct dependency
+        return StringUtils.contains(rawDependency, "specified in") || StringUtils.contains(rawDependency, "workspace-aggregator-");
+    }
+
+    /**
+     * Extracts a single dependency path from a raw dependency Json string returned from 'Yarn why' command.
+     *
+     * @param projectRootId   - The name of the project to display in the root of the impact tree.
+     * @param packageFullName - The vulnerable dependency in <NAME>:<VERSION> format.
+     * @param rawDependency   - The raw dependency Json string returned from 'Yarn why' command.
+     * @return The extracted dependency path as a list of dependencies starting from projectRootId till packageFullName.
+     */
+    private List<String> extractSinglePath(String projectRootId, String packageFullName, String rawDependency) {
+        List<String> pathResult = new ArrayList<>();
+        pathResult.add(projectRootId); // The root project is guaranteed to be the first element in the path
+        // remove any "_project_" strings (can be generated as part of a Yarn workspace in Yarn Monorepo feature)
+        rawDependency = StringUtils.remove(rawDependency, "_project_");
+
+        if (isDirectDependency(rawDependency)) { // This is a direct dependency
+            pathResult.add(packageFullName);
+            return pathResult;
+        }
+
+        // Split the path by '#'
+        String[] splitPath = StringUtils.split(StringUtils.substringBetween(rawDependency, "\""), "#");
+
+        if (splitPath == null) {
+            return null;
+        }
+
+        // packageFullName is guaranteed to be the last element in the path
+        if (!StringUtils.equals(splitPath[splitPath.length - 1], (StringUtils.substringBefore(packageFullName, ":")))) {
```

Comment:
Redundant parenthesis
```suggestion
        if (!StringUtils.equals(splitPath[splitPath.length - 1], StringUtils.substringBefore(packageFullName, ":"))) {
```